test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+      end
+      it "writes cronfile to sync ntp via cron" do
+        subject.synchronize_time = true
+        expect(Yast::SCR).to receive(:Write)
+          .with(Yast::Path.new(".target.string"), subject.cron_file, cron_entry)
+
+        subject.Write
+      end
+    end
+
+    it "returns true if not aborted" do
+      expect(subject.Write).to eql(true)
+    end
+  end
+
+  describe "#GetNtpServers" do
+    let(:country_servers) do
+      { "tick.nap.com.ar" =>
+        { "access_policy "  => "open access, please send a message to notify",
+          "address"         => "tick.nap.com.ar",
+          "country"         => "AR",
+          "exact_location"  => "Network Access Point, Buenos Aires, Argentina",
+          "location"        => "Argentina",
+          "stratum"         => "2",
+          "synchronization" => "NTP V3 secondary (stratum 2), Cisco IOS"
+        }
+      }
+    end
+
+    context "when ntp servers haven't been read before" do
+      it "returns empty hash in test Mode" do
+        allow(Yast::Mode).to receive(:test).and_return(true)
+
+        expect(subject.GetNtpServers).to eql({})
+      end
+
+      it "caches known ntp servers" do
+        allow(Yast::Mode).to receive(:test).and_return(false)
+        subject.instance_variable_set(:@ntp_servers, nil)
+        expect(subject).to receive(:update_ntp_servers!)
+
+        subject.GetNtpServers
+      end
+
+      it "return known ntp servers" do
+        subject.instance_variable_set(:@ntp_servers, country_servers)
+
+        expect(subject.GetNtpServers).to eql(country_servers)
+      end
+    end
+
+    context "when ntp servers haven been read before" do
+
+      before do
+        subject.instance_variable_set(:@ntp_servers, country_servers)
+      end
+
+      it "returns known ntp servers" do
+        expect(subject.GetNtpServers).to eql(country_servers)
+      end
+    end
+
+  end
+
+  describe "#reachable_ntp_server?" do
+    context "given a server" do
+      it "returns true if sntp test passed with IPv4 or/and with IPv6" do
+        allow(subject).to receive(:sntp_test).with("server").and_return(true, false, true)
+        allow(subject).to receive(:sntp_test).with("server", 6).and_return(true, true)
+
+        expect(subject.reachable_ntp_server?("server")).to eql(true)
+        expect(subject.reachable_ntp_server?("server")).to eql(true)
+        expect(subject.reachable_ntp_server?("server")).to eql(true)
+      end
+
+      it "returns false if sntp test fails with IPv4 or/and with IPv6" do
+        allow(subject).to receive(:sntp_test).with("server").and_return(false)
+        allow(subject).to receive(:sntp_test).with("server", 6).and_return(false)
+
+        expect(subject.reachable_ntp_server?("server")).to eql(false)
+      end
+    end
+
+  end
+
+  describe "#sntp_test" do
+    let(:ip_version) { 4 }
+    let(:server) { "sntp.server.de" }
+    let(:output) { { "stdout" => "", "stderr" => "", "exit" => 0 } }
+
+    context "given a server" do
```

Comment:
Maybe this context is not needed, as you don't have a `not given a server` one :)